### PR TITLE
chore(sf|quadbin): rewrite all quadbin functions from js to sql

### PR DESCRIPTION
**Pull request by luipir**
# Description

Shortcut

- Story: https://app.shortcut.com/cartoteam/story/320183
- Autolink: [sc-320183]

port all possible JS code to SQL

## Type of change

- Refactor

# Acceptance

Please describe how to validate the feature or fix

1. all old unit test (and new one) have to pass

# Basic checklist

- Good PR name
- Shortcut link
- Just one issue per PR
- GitHub labels
- Proper status & reviewers
- Tests
- Documentation

### CODE REVIEW

File: clouds/snowflake/modules/sql/quadbin/QUADBIN_FROMZXY.sql [hunk 1]
```diff
@@ -2,30 +2,78 @@
 -- Copyright (C) 2022 CARTO
```
Comment (vdelacruzb) [inline clouds/snowflake/modules/sql/quadbin/QUADBIN_FROMZXY.sql:2]: We have to update the implementation dates.

File: clouds/snowflake/modules/sql/quadbin/_FROM_BASE.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+----------------------------
+-- Copyright (C) 2023 CARTO
+----------------------------
+
+-- change a float or numeri to other base
```
Comment (vdelacruzb) [inline clouds/snowflake/modules/sql/quadbin/_FROM_BASE.sql:5]: `numeric`

Don't know if the name is too generic a and could collide with any other future UDF in the toolbox.

File: clouds/snowflake/modules/sql/quadbin/_TO_BASE.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+----------------------------
+-- Copyright (C) 2023 CARTO
+----------------------------
+
+-- change a float or numeri to other base
+CREATE OR REPLACE FUNCTION @@SF_SCHEMA@@._TO_BASE
```
Comment (vdelacruzb) [inline clouds/snowflake/modules/sql/quadbin/_TO_BASE.sql:6]: Same that `_FROM_BASE`

File: clouds/snowflake/modules/sql/quadbin/QUADBIN_BBOX.sql [hunk 4]
```diff
@@ -1,34 +1,50 @@
 ----------------------------
--- Copyright (C) 2022 CARTO
+-- Copyright (C) 2023 CARTO
```
Comment (vdelacruzb) [inline clouds/snowflake/modules/sql/quadbin/QUADBIN_BBOX.sql:2]: The dates should be 2022-2023 as we do in the branch `chore/sc-320182/optimize-quadbin-polyfill-snowflake`

**Comment by luipir** [comment]
creating the CTE as temporary table solve the issue (with almost similar execution time at elast for BBOX) but we loose the use of UDF and need to use procedures.

no idea if it is necessary to continue or assume the ticket blocked waiting for SF fixes of different strategies.

**Comment by Jesus89** [comment]
Superseded by https://github.com/CartoDB/analytics-toolbox-core/pull/423

**Comment by luipir** [comment]
sorry closed too early... effectively all this JS function are not jet available in https://github.com/CartoDB/analytics-toolbox-core/pull/423

**Pull request closed without merging**